include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -8946,6 +8946,9 @@ def err_bounds_type_annotation_lost_checking : Error<
   def err_checked_scope_no_variadic_func_for_expression : Error<
     "cannot use a variable arguments function in a checked scope or function">;
 
+  def err_checked_scope_no_assume_bounds_casting : Error<
+    "invalid _Assume_bounds_cast in a checked scope or function">;
```

Comment:
I think it would be better to say "_Assume_bounds_cast not allowed in a checked scope or function".    Otherwise programmers may think there is something wrong with the _Assume_bounds_cast expression itself.